ruby.asciidoc

```diff
@@ -0,0 +1,565 @@
+= Pony for Rubyists
+
+In this chapter we’ll take a look at Pony from a Ruby programmer’s
+perspective, with the assumption that the reader knows Ruby.
+
+[[what-we-like-about-ruby]]
+What we like about Ruby
+-----------------------
+
+As Ruby developers, we like Ruby’s simple design and approachable
+syntax. It has a strong focus on developer happiness and it's easy for
+newcomers to get up and running quickly.
+
+Ruby's simplicity allows developers to switch between Object-oriented
+and vaguely functional styles with ease. Blocks and dynamic typing allow
+Rubyists to be incredibly expressive in how they approach problems.
+
+Ruby also comes with a broad standard library that has provided a
+foundation for the community to build a rich ecosystem of libraries that
+cover a wide range of domains.
+
+[[how-pony-is-like-ruby]]
+How Pony is like Ruby
+---------------------
+
+Pony has a number of attributes that are similar to Ruby's, and in most
+cases a bit better too. Some of these are:
+
+* Pony is also simple and approachable
+* Pony is also object oriented, plus it blends in functional elements
+* Pony is also garbage collected, eliminating the need for developers to
+manually track memory use
+
+Other aspects of Pony feel quite a bit different than Ruby's:
+
+* Pony is strongly and statically typed, with features like algebraic
+types and traits
+* Pony’s concurrency model is a strong point, using actors with
+asynchronous behaviors that safely encapsulate all concurrent operations
+* Pony’s `ponyc` compiler is very fast, and compiles to native code
+using the LLVM backend which allows it to target just as many platforms
+if not more than Ruby
+* Pony uses distinct keywords, with `var` to introduce variables,
+and `let` to introduce values
+
+[[installing-pony]]
+Installing Pony
+---------------
+
+All details can be found in the
+https://github.com/ponylang/ponyc/blob/master/README.md#installation[installation
+instructions].
+
+[[learning-by-example]]
+Learning by Example
+-------------------
+
+Let’s get to know Pony a little better by walking through and comparing
+some examples written in both Ruby and Pony.
+
+[[hello-world]]
+Hello World
+~~~~~~~~~~~
+
+We’ll begin by comparing the ubiquitous "hello world" in both languages.
+
+Here’s a minimal example in Ruby:
+
+[source,ruby]
+----
+$ cat hello_world.rb
+
+puts "hello world!"
+----
+
+To run the program, execute these commands in the helloworld directory:
+
+[source,shell]
+----
+ruby hello_world.rb
+----
+
+In Ruby, the source file name can be anything, and can contain as many
+classes as you like.
+
+Now for the pony version:
+
+[source,ponylang]
```

Comment:
I'm not sure about gitbook, but to get the source code to render correctly in github, the language name should be `pony` instead of `ponylang`.  Same comment applies to all occurrences in this chapter.